### PR TITLE
Feature/5752 constrain pools table by teams

### LINKED ISSUE #5752: ✨ Pools table should only show pools user has access to
## ✨ Feature

The pools table on Admin currently lists all pools. However, we're moving to a model where the page is only intended for Pool Operators, who always belong to a team. A user may in fact be a pool operator in multiple teams, but they should not be able to see Pools from any other teams.

## 🕵️ Details

> __Note__
> This assumes only Pool Operators can view this page. If we want to make the Pool Table available to other roles (who don't have teams) we'll need to either create distinct pages (My Pools vs All Pools?) or change the query based on the user's role.

## 🎨 Design File

Include a link to the design file (if it exists).

## 📸 Screenshot

Add a screenshot of the new feature (if possible).

## 🙋‍♀️ Proposed Implementation

Instead of querying for all teams, the page should should query for `me.roleAssignments.team.pools`, getting all pools belonging to all the teams a user is in, and combining those into a single collection before passing to the table component.

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] The admin Pools table only lists pools belonging to teams the user is a part of.

## :stop_sign: Blockers

List any issues which must be completed before this one.

```[tasklist]
### Blockers
- [ ] #5563 
```

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #5752 

## 👋 Introduction

Rather than fetch all pools for the table, only pass in what the user is able to view.

## 🕵️ Details

Swapping queries and some type wrangling 😖 

## 🧪 Testing

1. go to /pools and affirm you can only see the pools you ought to see
2. run this query in playground to see what pools you have access to
```

query getMyPools {
  me {
    roleAssignments {
      team {
        id 
        name
        pools {
          id 
          name {
            en
          }
        }
      }
    }
  }
}
```

## 📸 Screenshot

image

image

### CODE REVIEW

File: apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx [hunk 1]
```diff
@@ -357,12 +365,48 @@ export const PoolTable = ({ pools }: PoolTableProps) => {
 };
 
 const PoolTableApi = () => {
-  const [result] = useGetPoolsQuery();
+  // data manipulation functions
```
Comment (petertgiles) [inline apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx:368]: These two functions could be moved outside the component so they don't get recreated on every render?

**Review by petertgiles — COMMENTED** [review]
I need to test this more but some quick comments...

File: apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx [hunk 2]
```diff
@@ -357,12 +365,48 @@ export const PoolTable = ({ pools }: PoolTableProps) => {
...
+  // roles assignments to team collection
+  const createTeamsArray = (
```
[the remark this answers is not in the document]
Comment (vd1992) [reply apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx]: updated things a good bit in https://github.com/GCTC-NTGC/gc-digital-talent/pull/5870/commits/0ecca046d701c3f4147e758f25126ddffc4d5961 and https://github.com/GCTC-NTGC/gc-digital-talent/pull/5870/commits/a3df9b938ce46162c6e687f857ae0f761b406201
used flatmap and seems to work alright

**Review by petertgiles — APPROVED** [review]
Looks great!

**Pull request merged**